### PR TITLE
[SPARK-10577] [PySpark] DataFrame hint for broadcast join

### PR DESCRIPTION
**Pull request by snowmoon-zhang**
https://issues.apache.org/jira/browse/SPARK-10577

### CODE REVIEW

**Comment by rxin** [comment]
Jenkins, test this please.

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -189,6 +190,14 @@ def approxCountDistinct(col, rsd=None):
     return Column(jc)
 
 
+@since(1.6)
+def broadcast(df):
+    """Marks a DataFrame as small enough for use in broadcast joins."""
+
+    sc = SparkContext._active_spark_context
+    return DataFrame(sc._jvm.functions.broadcast(df._jdf), sc._jsc)
```
Comment (davies) [inline python/pyspark/sql/functions.py]: `DataFrame(sc._jvm.functions.broadcast(df._jdf), df.sql_ctx)`

Comment (Jianfeng-chs) [inline python/pyspark/sql/functions.py]: @davies changed it. Thank you!

**Comment by Jianfeng-chs** [comment]
@rxin Sorry for inconvenience. Can you trigger a new test for this. I have merged changes into one single commit. Thanks

**Comment by rxin** [comment]
Thanks - I've merged this.

**Comment by rxin** [comment]
@Jianfeng-chs  what's your JIRA user id? I need that in order to assign the ticket to you.

**Comment by Jianfeng-chs** [comment]
Thanks @rxin. My JIRA username is jzhang.

**Pull request closed without merging**